### PR TITLE
Fix upload/download

### PR DESCRIPTION
**Pull request by Fosol**
The upload/download was broken when the data source dropdown list was changed.  I've modified the design so that this won't happen, and also solved other issues I was aware of.

- Uploaded file path is now stored so that it doesn't change after first created
- Changing the data source after won't break the file

> Note that existing files already uploaded will probably not work.  I don't want to spend time moving them around at this point.

### CODE REVIEW

File: app/editor/src/components/upload/Upload.tsx [hunk 1]
```diff
@@ -103,7 +103,7 @@ export const Upload: React.FC<IUploadProps> = ({
         hide={toggle}
         type="delete"
         headerText="Confirm Removal"
-        body="Are you sure you want to remove this file?"
+        body="Are you sure you want to remove this file?  You will still need to save before it is deleted."
```
Comment (Fosol) [inline app/editor/src/components/upload/Upload.tsx:106]: Clarity

File: app/editor/src/features/content/form/ContentForm.tsx [hunk 2]
```diff
@@ -127,7 +127,6 @@ export const ContentForm: React.FC<IContentFormProps> = ({ contentType = Content
       // If the upload fails, we still need to update the form from the original update.
       if (!!contentResult) {
         setContent(toForm(contentResult));
-        toast.error(`${contentResult.headline} file was not uploaded successfully.`);
```
Comment (Fosol) [inline app/editor/src/features/content/form/ContentForm.tsx:130]: No need for another error

File: app/editor/src/features/content/form/ContentForm.tsx [hunk 3]
```diff
@@ -197,9 +196,9 @@ export const ContentForm: React.FC<IContentFormProps> = ({ contentType = Content
                           ''
                         }
                         onChange={(newValue: any) => {
-                          props.setFieldValue('dataSourceId', newValue.value);
-                          props.setFieldValue('source', newValue.label);
                           const dataSource = dataSources.find((ds) => ds.id === newValue.value);
+                          props.setFieldValue('dataSourceId', newValue.value);
+                          props.setFieldValue('source', dataSource?.code ?? '');
```
Comment (Fosol) [inline app/editor/src/features/content/form/ContentForm.tsx:201]: Use the code, not the value from the dropdown label

File: libs/net/dal/Extensions/FileReferenceExtensions.cs [hunk 4]
```diff
@@ -39,7 +53,7 @@ public static string GetFilePath(this IReadonlyFileReference fileReference, TNOC
         // TODO: Handle when the original file uploaded has different path than the new one.
         var dataConnection = JsonSerializer.Deserialize<Dictionary<string, object>>(location.Connection) ?? new Dictionary<string, object>();
         var connectionPath = dataConnection.ContainsKey("path") ? $"{((string?)dataConnection["path"])?.RemoveStartAndEnd("/")}" : "";
-        var path = Path.Combine(storageConfig.GetPath(), source ?? "", connectionPath);
```
Comment (Fosol) [inline libs/net/dal/Extensions/FileReferenceExtensions.cs:42]: Don't include the source as part of the storage location

File: libs/net/dal/Migrations/TNOContextModelSnapshot.cs [hunk 5]
```diff
@@ -121,7 +121,7 @@ protected override void BuildModel(ModelBuilder modelBuilder)
                     b.HasIndex("Name")
                         .IsUnique();
 
-                    b.ToTable("action");
+                    b.ToTable("action", (string)null);
```
Comment (Fosol) [inline libs/net/dal/Migrations/TNOContextModelSnapshot.cs]: Hmm this doesn't look right.

**Review by jdtoombs — APPROVED** [review]

File: libs/net/dal/Models/ContentFileReference.cs [hunk 6]
```diff
@@ -124,7 +159,7 @@ public ContentFileReference(FileReference fileReference, IFormFile file)
         this.File = file ?? throw new ArgumentNullException(nameof(file));
         this.Id = fileReference.Id;
         this.FileName = file.FileName;
-        this.Path = GenerateFileName(this.Content, file);
+        this.Path = GenerateFilePath(this.Content, file);
```
Comment (Fosol) [inline libs/net/dal/Models/ContentFileReference.cs:162]: Keep the whole path to the file (not including the data location config path).

File: libs/npm/core/src/hooks/summon/useSummon.tsx [hunk 7]
```diff
@@ -20,14 +20,10 @@ let axiosInstances = {
 /**
  * useSummon hook properties.
  */
-export interface ISummonProps {
+export interface ISummonProps extends AxiosRequestConfig<any> {
```
Comment (Fosol) [inline libs/npm/core/src/hooks/summon/useSummon.tsx:23]: Allow us to configure summon/axios going forward.

**Pull request merged**